src/main/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandler.java

```diff
@@ -547,90 +506,80 @@ private void setConfiguration(RoutingContext ctx) {
     }
 
     private void getMonitorInformation(RoutingContext ctx) {
-        final boolean emptyQueues = evaluateUrlParameterToBeEmptyOrTrue(EMPTY_QUEUES_PARAM, ctx.request());
-        final int limit = extractLimit(ctx);
-        String filter = ctx.request().params().get(FILTER);
-        eventBus.request(redisquesAddress, buildGetQueuesItemsCountOperation(filter), (Handler<AsyncResult<Message<JsonObject>>>) reply -> {
-            if (reply.succeeded() && OK.equals(reply.result().body().getString(STATUS))) {
-                JsonArray queuesArray = reply.result().body().getJsonArray(QUEUES);
-                if (queuesArray != null && !queuesArray.isEmpty()) {
-                    List<JsonObject> queuesList = queuesArray.getList();
-                    queuesList = sortJsonQueueArrayBySize(queuesList);
-                    if (!emptyQueues) {
-                        queuesList = filterJsonQueueArrayNotEmpty(queuesList);
-                    }
-                    if (limit > 0) {
-                        queuesList = limitJsonQueueArray(queuesList, limit);
-                    }
+        queueStatsService.getQueueStats(ctx, queueStatsMentor);
+    }
 
-                    // this function always succeeds, no need to handle the error case
-                    fillStatisticToQueuesList(queuesList).onSuccess(updatedQueuesList -> {
-                        JsonObject resultObject = new JsonObject();
-                        resultObject.put(QUEUES, updatedQueuesList);
-                        jsonResponse(ctx.response(), resultObject);
-                    });
-                } else {
-                    // there was no result, we as well return an empty result
-                    JsonObject resultObject = new JsonObject();
-                    resultObject.put(QUEUES, new JsonArray());
-                    jsonResponse(ctx.response(), resultObject);
-                }
-            } else {
-                String error = "Error gathering names of active queues";
-                log.error(error, reply.cause());
-                respondWith(StatusCode.INTERNAL_SERVER_ERROR, error, ctx.request());
+    private class MyQueueStatsMentor implements GetQueueStatsMentor<RoutingContext> {
+
+        @Override
+        public boolean includeEmptyQueues(RoutingContext ctx) {
+            return evaluateUrlParameterToBeEmptyOrTrue(EMPTY_QUEUES_PARAM, ctx.request());
+        }
+
+        @Override
+        public int limit(RoutingContext ctx) {
+            return extractLimit(ctx);
+        }
+
+        @Override
+        public String filter(RoutingContext ctx) {
+            return ctx.request().params().get(FILTER);
+        }
+
+        @Override
+        public void onQueueStatistics(List<QueueStatsService.Queue> queues, RoutingContext ctx) {
+            var rsp = ctx.request().response();
+            rsp.putHeader(CONTENT_TYPE, APPLICATION_JSON);
+            rsp.setChunked(true);
+            rsp.write("{\"queues\": [");
```

Comment:
why not write the hole response as one JsonObject instead of adding error prone, manual start and end strings?

Reply:
Because I prefer to instantiate ONE SINGLE JsonObject instance over producing several thousands of them.